snuba/subscriptions/executor.py

```diff
@@ -18,9 +23,25 @@ class SubscriptionExecutor:
     to make the query, and the result is then returned as a `Future`.
     """
 
-    def __init__(self, dataset: Dataset, executor_pool: ThreadPoolExecutor):
+    def __init__(
+        self,
+        dataset: Dataset,
+        executor_pool: ThreadPoolExecutor,
+        metrics: MetricsBackend,
+    ):
         self.__dataset = dataset
         self.__executor_pool = executor_pool
+        self.__metrics = metrics
+        self.__concurrent_gauge = Gauge(self.__metrics, "executor.concurrent")
+
+    def __run_query(
+        self, scheduled_at: datetime, request: Request, timer: Timer
+    ) -> ClickhouseQueryResult:
+        self.__metrics.timing(
+            "executor.latency", (time.time() - scheduled_at.timestamp()) * 1000,
+        )
+        with self.__concurrent_gauge:
```

Comment:
This is likely going to be rolled up to the last observation over the submission window so is not wildly accurate, but over time should be enough to figure out if we're fully utilizing the thread pool.